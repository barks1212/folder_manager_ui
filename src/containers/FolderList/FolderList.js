import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { folderDataType } from '../../types/types';

import Folder from '../../components/Folder/Folder';
import File from '../../components/File/File';


class FolderList extends Component {
    state = {
        open: false
    }
   

    

    clickHandler(event, open, fileName) {
        event.preventDefault();
        this.setState({open: !open})

        
    };

    render() {
        const { data } = this.props;

        const renderFileOrFolder = data.map(contents => (
            contents.type === 'folder' ?
             <Fragment>
                <div data-testid="folder">
                    <Folder key={contents.name} contents={contents}/>
                </div>
            </Fragment> :
            <Fragment>
                <div data-testid="file">
                    <File key={contents.name} contents={contents} />
                </div>
            </Fragment>
        ));

        return (
            <div>
                {renderFileOrFolder}
            </div>
        );
    };
};

const mapStateToProps = state => ({
    data: state.folderData,
})

FolderList.propTypes = {
    data: PropTypes.arrayOf(PropTypes.shape({
        ...folderDataType,
        files: PropTypes.arrayOf(folderDataType)
    }))
};

FolderList.defaultProps = {
    data: []
};

export default connect(mapStateToProps, null)(FolderList);