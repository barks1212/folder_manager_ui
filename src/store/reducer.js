import folderData from '../data/folders.json';
import * as actionTypes from './actions';

const initialState = {
    folderData,
    sort: null,
}

const reducer = (state = initialState, action) => {
    switch (action.type) {
        case actionTypes.SET_SORT:
            if (action.sort === 'name') {
                return {
                    ...state,
                    sort: 'name'
                }
            }
            if (action.sort === 'date') {
                return {
                    ...state,
                    sort: 'date'
                };
            };
            break;
        case actionTypes.SORT_FILES:
            let sortedState;
            if (action.sort === 'name') {
                sortedState = folderData.sort((a, b) => {
                    if (a.name < b.name) return -1;
                    if (a.name > b.name) return 1;
                    return 0;
                });
                return {
                    ...state,
                    folderData: sortedState
                };
            };
            break;
        default:
            return state;
    };
};

export default reducer;