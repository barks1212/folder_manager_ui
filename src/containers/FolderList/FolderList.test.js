import React from 'react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { render, cleanup } from '@testing-library/react'

import FolderList from './FolderList';
import reducer from '../../store/reducer';
import data from '../../data/folders.json';

const store = createStore(reducer)

afterEach(cleanup);

describe('folderList', () => {
    it('should render', () => {
        const {asFragment} = render(
        <Provider store={store}>
            <FolderList />
        </Provider>)
        expect(asFragment()).toMatchSnapshot();
    })

    it('renders a list of files and folders', () => {
        const {getAllByTestId} = render(
        <Provider store={store}>
            <FolderList />)
        </Provider>
        );
        const files = getAllByTestId('file').map(f => f.firstElementChild.firstChild.childNodes[1].textContent);
        const folders = getAllByTestId('folder').map(f => f.firstElementChild.firstChild.childNodes[0].textContent);

        const filesToMatch = data.filter(el => el.type !== 'folder').map(f => f.name);
        const foldersToMatch = data.filter(el => el.type === 'folder').map(f => f.name);

        expect(files).toEqual(filesToMatch);
        expect(folders).toEqual(foldersToMatch);
    });
});