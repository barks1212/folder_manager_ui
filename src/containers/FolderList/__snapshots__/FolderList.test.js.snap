// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`folderList should render 1`] = `
<DocumentFragment>
  <div>
    <div
      data-testid="file"
    >
      <div>
        <ul>
          <li>
            pdf
          </li>
          <li>
            Employee Handbook
          </li>
          <li>
            2017-01-06
          </li>
        </ul>
      </div>
    </div>
    <div
      data-testid="file"
    >
      <div>
        <ul>
          <li>
            pdf
          </li>
          <li>
            Public Holiday policy
          </li>
          <li>
            2016-12-06
          </li>
        </ul>
      </div>
    </div>
    <div
      data-testid="folder"
    >
      <div>
        <ul>
          <li>
            folder
          </li>
          <li>
            Expenses
          </li>
        </ul>
      </div>
    </div>
    <div
      data-testid="file"
    >
      <div>
        <ul>
          <li>
            csv
          </li>
          <li>
            Cost centres
          </li>
          <li>
            2016-08-12
          </li>
        </ul>
      </div>
    </div>
    <div
      data-testid="folder"
    >
      <div>
        <ul>
          <li>
            folder
          </li>
          <li>
            Misc
          </li>
        </ul>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
